### DISCUSSION TITLE
Map tidak muncul di streamlit python

### DISCUSSION THREAD

**Discussion by yohanespm77** [teamalgoritma/community#710] [Q&A] [answered]
Saya punya kodingan seperti ini :

#file peta.py
<img width="646" alt="image" src="https://github.com/teamalgoritma/community/assets/106010284/60081a97-baeb-4009-993f-b80884a95508">

#file main.py
<img width="443" alt="image" src="https://github.com/teamalgoritma/community/assets/106010284/1788fc05-763f-43aa-a54f-f9868933b79f">

Namun saat dijalankan, mapnya tidak muncul 
<img width="842" alt="image" src="https://github.com/teamalgoritma/community/assets/106010284/e875fefb-7f75-4b3f-9e48-8cd070ddad07">

Mohon bantuannya kak, terima kasih 🙏

**Comment by IrfanChairurrachman** [comment]
Dear @yohanespm77 ,

Jika fungsi dan peta `st_folium` dipanggil di app.py alih-alih dipanggil melalui peta.py apakah bisa Pak? Berhubung dari dokumentasi resmi streamlit folium https://folium.streamlit.app/static_map menggunakan `st_folium` di app.py. 

Ini biasa terjadi khususnya di peta interaktif karena beberapa fungsi untuk merender peta interaktif tidak akan melakukan render langsung karena hasil dipanggil dari fungsi file luar.

**Reply by yohanespm77** [reply]
kalau peta.py nya berdiri sendiri dan tidak masuk dalam def load_view() bisa mas. Tapi saya ingin peta.py nya terhubung ke main.py

**Reply by IrfanChairurrachman** [reply]
Jika mmng ingin objek petanya dari `load_view()` mungkin bisa direturn objek folium `m`nya saja Pak. Kemudian untuk merendernya baru menggunakan fungsi `st_folium` di app.py. Jadi kira kira seperti ini:

peta.py
```python
def load_view():
    # center on Liberty Bell, add marker
    m = folium.Map(location=[39.949610, -75.150282], zoom_start=16)
    folium.Marker(
        [39.949610, -75.150282], popup="Liberty Bell", tooltip="Liberty Bell"
    ).add_to(m)

    return m
```

app.py
```python
from views import peta

m_object = peta.load_view()
st_folium(m_object, width=725, returned_objects=[])
```

Semoga informasi ini dapat membantu. Bila ternyata informasi ini dapat menjawab pertanyaan Anda, Anda dapat “Mark as answer” jawaban ini.

**Reply by yohanespm77** [reply]
Ga muncul juga kak @IrfanChairurrachman 

Sudah saya buat seperti ini : 

peta.py
<img width="535" alt="image" src="https://github.com/teamalgoritma/community/assets/106010284/0f1a4eca-f24b-46fe-8340-f7a0baff5a4a">

main.py
<img width="469" alt="image" src="https://github.com/teamalgoritma/community/assets/106010284/87a336e3-20db-4a2c-aa55-45fc548d18d8">


Hasilnya : 
<img width="912" alt="image" src="https://github.com/teamalgoritma/community/assets/106010284/f386f3f5-79f8-4b16-bba7-48e7dbfd539c">

Masih belum muncul mapnya kak

**Reply by IrfanChairurrachman** [reply, marked as answer]
Konsepnya sama Pak seperti permasalahan sebelumnya seperti memanggil fungsi `load_view()`, di sini Bapak masih memasukkan `st_folium()` di dalam fungsi `navigation()` (walaupun masih di dalam app.py). Sehingga Bapak kode untuk render folium tidak dibalut oleh fungsi apapun.

Jika Bapak melihat contoh-contoh projek yang lain menggunakan folium streamlit, tidak ada yang menggunakan fungsi untuk merender (seperti st_folium) yang dibalut oleh fungsi lagi. Mungkin salah satu contohnya seperti berikut: https://github.com/randyzwitch/streamlit-folium/tree/master/examples 

Namun, jika Bapak ingin menggunakan cara lain untuk menampilkan folium di streamlit tanpa menggunakan `st_folium` dapat mengacu pada [artikel berikut](https://medium.com/analytics-vidhya/deploying-your-geospatial-machine-learning-projects-as-web-apps-using-streamlit-and-heroku-45d64f6d5cb0) dimana alih-alih menggunakan `st_folium` untuk merender peta, dia menggunakan `m._repr_html_()` untuk menampilkan peta.

**Reply by yohanespm77** [reply]
Navigationnya itu untuk multipagenya mas. Jadi untuk page "Peta", akan menampilkan map. Untuk page "Trendline" akan menampilkan plot. 
Referensi saya utk membuat multipage darisini : https://github.com/BugzTheBunny/streamlit-navbar-flaskless

Saya coba pakai repr_html() tampilannya malah seperti ini : 
<img width="880" alt="image" src="https://github.com/teamalgoritma/community/assets/106010284/fc6dff8e-8bbb-4787-80ae-8120b07ff72c">


Solusinya bagaimana mas ? @IrfanChairurrachman
